Tests/NIOTests/ChannelPipelineTest.swift

```diff
@@ -1294,6 +1294,141 @@ class ChannelPipelineTest: XCTestCase {
         XCTAssertNoThrow(try removal1Future.wait())
         XCTAssertNoThrow(XCTAssertTrue(try channel.finish().isClean))
     }
+
```

Comment:
I think we should add an alloc counter test that shows that it's actually cheaper :)

Reply:
Will do, wanted to get consensus on the changes first.

Reply:
Added a sync counterpart to the 'add two handlers 1000 times' and async and sync versions of 'get a typed handler from the pipeline'. Reckon we need any more?

Reply:
no, that's fine, thank you.